packages/node_modules/pouchdb-checkpointer/src/index.js

```diff
@@ -133,9 +133,21 @@ var comparisons = {
 
 Checkpointer.prototype.getCheckpoint = function () {
   var self = this;
+
+  if (self.opts && self.opts.writeSourceCheckpoint && !self.opts.writeTargetCheckpoint) {
```

Comment:
I wonder whether this should be combined with `self.readOnlySource` - it seems like we have two properties to express the same thing

Reply:
The way I read it, there are four different options:

`source` \ `target` | on | off
---|---|---
**on** | default | previously broken
**off** | previously broken | no checkpointing

How would you recommend combining?

Reply:
`self.readOnlyCheckpoint` is set when we detect that we don't have write permissions on the source. It seems like, once set, we'd want the behaviour to be the same as if `opts.writeSourceCheckpoint` is `false`. Looking at it again, this line probably doesn't change, but we have 2 similar-looking flags adjusting checkpoint behaviour that look like they should be combined in general

Reply:
I'm not sure it's that simple, though.  One flag is a request for behaviour, and the other is a reaction to discovered properties of a DB.  E.g.
```
Checkpointer.prototype.updateSource = function (checkpoint, session) {
  if (this.opts.writeSourceCheckpoint) {
    var self = this;
    if (this.readOnlySource) {
      return Promise.resolve(true);
    }
```
Perhaps the auto-detection could be removed and an error thrown if the source is discovered to be read only when the user has requested checkpointing on both sides?  OTOH the current behaviour is quite helpful...

Reply:
@willholley PR for combining the two at https://github.com/pouchdb/pouchdb/pull/6738